package/MDAnalysis/coordinates/DLPoly.py

```diff
@@ -216,31 +227,17 @@ def _read_frame(self, frame):
 
     @property
     def n_frames(self):
-        try:
-            return self._n_frames
-        except AttributeError:
-            self._n_frames = self._read_n_frames()
-            return self._n_frames
-
-    def _read_n_frames(self):
-        """Read the number of frames, and the offset for each frame
-
-        offset[i] - returns the offset in bytes to seek into the file to be
-                    just before the frame starts
-        """
-        offsets = self._offsets = []
+        # Second line is traj_key, imcom, n_atoms, n_frames, n_records
+        offsets = []
 
         with open(self.filename, 'r') as f:
```

Comment:
The reason for `self._n_frames` was to cache the number of frames so it's only calculated once.  This needs to be added back in.

Reply:
Actually, a question about the expected behaviour if a *wrong* HISTORY file is passed. DL_POLY HISTORY files have the `n_frames` added to the header line on top, so from the test files `DLP_HISTORY_minimal`:

```
DL_POLY: Minimal example of HISTORY
         0         0       3                    3                 2606
```

it can be parsed as:
*trajectory key:* `0` - only coordinates saved
*imcom:* `0` - non periodic
*n_atoms:* `3` - three atoms
*n_frames:* `3` - three frames
*n_records:* `2606` - number of records (typically lines, wrong here).

The `HistoryReader()` takes `n_atoms` from there. I also made it take `n_frames` from there, but it might be that someone just truncated the trajectory file and the information might be wrong. Should we instead just go through the file updating the number of frames as it reads? That is:

`n_frames = len(offsets)`.

I think it might be more bullet proof, as I can totally imagine when someone would remove some last/first 50 frames without updating a header, but not touch the number of atoms at all.

Reply:
Re caching, I don't think the `HistoryReader()` was doing that but it makes sense to add, thanks!